template/template.go

```diff
@@ -200,18 +255,33 @@ func (e *executer) resolveDeclarations(declarations []ast.Declaration) error {
 		case ast.ReservedStatement:
 			return fmt.Errorf("%w", mf2.ErrUnsupportedStatement)
 		case ast.LocalDeclaration:
-			e.variables[string(d.Variable)] = newVariable(d.Expression, nil)
+			r, err := e.resolveExpression(d.Expression)
+			if err != nil {
+				r.err = errors.Join(r.err, fmt.Errorf("resolve for %s: %w", string(d.Variable), err))
+			}
+
+			e.variables[string(d.Variable)] = r // newVariable(d.Expression, nil)
 		case ast.InputDeclaration:
 			name := string(d.Operand.(ast.Variable)) //nolint: forcetypeassert // always ast.Variable
 
 			val, ok := e.variables[name]
 			if !ok {
-				return mf2.ErrUnresolvedVariable
+				r := NewResolvedValue("{$" + name + "}")
```

Comment:
Would it make sense to create a helper function?

```suggestion
				r := newFallbackValue(name)
```